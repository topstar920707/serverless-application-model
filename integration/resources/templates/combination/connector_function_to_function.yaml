Resources:
  MyRole1:
    Type: AWS::IAM::Role
    Properties:
      AssumeRolePolicyDocument:
        Statement:
          - Effect: Allow
            Action: sts:AssumeRole
            Principal:
              Service: lambda.amazonaws.com
      ManagedPolicyArns:
        - !Sub arn:${AWS::Partition}:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole

  MyRole2:
    Type: AWS::IAM::Role
    Properties:
      AssumeRolePolicyDocument:
        Statement:
          - Effect: Allow
            Action: sts:AssumeRole
            Principal:
              Service: lambda.amazonaws.com
      ManagedPolicyArns:
        - !Sub arn:${AWS::Partition}:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole

  TriggerFunction:
    Type: AWS::Lambda::Function
    Properties:
      Role: !GetAtt MyRole1.Arn
      Runtime: nodejs14.x
      Handler: index.handler
      Code:
        ZipFile: |
          const AWS = require('aws-sdk');
          exports.handler = async (event) => {
            const params = {
              FunctionName: process.env.FUNCTION_NAME,
              InvocationType: 'RequestResponse',
              Payload: '{}',
            };
            const lambda = new AWS.Lambda();
            const response = await lambda.invoke(params).promise();
            if(response.StatusCode !== 200){
              throw new Error('Failed to get response from lambda function')
            }
            return response;
          };
      Environment:
        Variables:
          FUNCTION_NAME: !Ref MyFunction2

  MyFunction2:
    Type: AWS::Lambda::Function
    Properties:
      Role: !GetAtt MyRole2.Arn
      Runtime: nodejs14.x
      Handler: index.handler
      Code:
        ZipFile: |
          const AWS = require('aws-sdk');
          exports.handler = async (event) => {
            console.log(JSON.stringify(event));
          };

  MyConnector:
    Type: AWS::Serverless::Connector
    Properties:
      Source: 
        Id: TriggerFunction
      Destination: 
        Id: MyFunction2
      Permissions:
        - Write
