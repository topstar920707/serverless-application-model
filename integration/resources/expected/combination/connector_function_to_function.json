[
  {
    "LogicalResourceId": "MyRole1",
    "ResourceType": "AWS::IAM::Role"
  },
  {
    "LogicalResourceId": "MyRole2",
    "ResourceType": "AWS::IAM::Role"
  },
  {
    "LogicalResourceId": "TriggerFunction",
    "ResourceType": "AWS::Lambda::Function"
  },
  {
    "LogicalResourceId": "MyFunction2",
    "ResourceType": "AWS::Lambda::Function"
  },
  {
    "LogicalResourceId": "MyConnectorPolicy",
    "ResourceType": "AWS::IAM::ManagedPolicy"
  }
]